src/components/navigationBar/NavigationBar.spec.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from "react";
+import "@testing-library/jest-dom/extend-expect";
+import { fireEvent, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { MenuItem } from "./MenuItem";
+import { NavigationBar } from "./NavigationBar";
+
+test("renders hamburger for expanding", () => {
+  const { container } = renderDummyNavBar();
+
+  expect(container.querySelector("svg")).toBeInTheDocument();
+});
+
+test("doesn't render hamburger when expanding is disabled", () => {
+  const { container } = renderDummyNavBar(false);
+
+  expect(container.querySelector("svg")).not.toBeInTheDocument();
+});
+
+test("renders MenuItems as icons without text when folded", () => {
+  const { container } = renderNavBarWithItems();
+
+  expect(container.querySelectorAll("svg")).toHaveLength(3);
+  expect(screen.queryByText("Dashboard")).toBeNull();
+});
+
+test("renders MenutItems as icons with text when unfolded", async () => {
+  const { container } = renderNavBarWithItems();
+
+  expect(screen.queryByText("Dashboard")).toBeNull();
+
+  const hamburger = container.querySelector("svg");
+
+  if (hamburger) {
+    userEvent.click(hamburger);
+  }
+
+  if (container.firstChild) {
+    fireEvent.transitionEnd(container.firstChild);
+  }
```

Comment:
I think it's better to make type assertion in these cases.
If something brakes, your if statement won't fire and the tests might pass.
With assertion it should throw an error.

Reply:
We still need those `ifs`. Without it `userEvent.click` throws type error as `hamburger` can be `null`.

<img width="823" alt="image" src="https://user-images.githubusercontent.com/2817128/153867173-d39a7e8f-48cc-4811-86e3-8fae37f998a8.png">

Nevertheless, I have added assertions for more readability in test report.

Reply:
Type assertion -> `userEvent.click(hamburger as SVGSVGElement)`
that's what i meant, this should remove the error